qemu/tests/x86_cpu_test_dies.py

```diff
@@ -0,0 +1,45 @@
+import random
+
+from avocado.utils import cpu
+
+from virttest import error_context
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check guest gets correct multiple vcpu dies
+
+    1) Boot guest with options: -smp n,dies=2x...
+    2) Check cpu dies(only for Linux guest and Intel host)
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vm_name = params['main_vm']
+    vcpu_dies_list = [2, 4]
+    params['vcpu_dies'] = random.choice(vcpu_dies_list)
+    params['start_vm'] = 'yes'
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    if params["os_type"] == "linux" and cpu.get_vendor() == 'intel':
+        check_die_id = params['check_die_id']
+        check_die_cpus_list = params['check_die_cpus_list']
+        vcpu_sockets = params['vcpu_sockets']
+        vcpu_dies = params['vcpu_dies']
```

Comment:
I suppose here `vm.cpuinfo.vcpu_sockets` and `vm.cpuinfo.vcpu_dies` would be better since the qemu command finally utilizes these two, and it probably be changed by qemu_vm.py when create qemu command line sometimes.

Reply:
> I suppose here `vm.cpuinfo.vcpu_sockets` and `vm.cpuinfo.vcpu_dies` would be better since the qemu command finally utilizes these two, and it probably be changed by qemu_vm.py when create qemu command line sometimes.

thanks and update.

Reply:
I have a little different opinion about this issue.
I meet the same problem before, and choose the params at last.  Because, `dies` is what user set for testing purpose, if it be changed in qemu_vm at last, means vm boot with another `dies`, may be not what we want, but we will miss this change since test still can pass.
If use params,  but it be changed in qemu_vm, it is true that params is not match with the qemu cli, test will fail and we need check and make sure the defined `dies` in the test case take effect in qemu cli.

Reply:
Em, thanks Yanan, I just basically think that this test case is to test if the dies information in the guest matches the qemu command line pass to it, so it does not really matter what `params['xx']` is in this case. Let feature owner decide it, both okay for me. @nanliu-r

Reply:
@yanan-fu @vivianQizhu Hi, seems there is not too much change for `die` and `sockets` in qemu_vm.py according to cfg. So I think both the way vm.cpuinfo is better and no inflence to `die` number.
and I add the parameter `vcpu_maxcpus = 0` in cfg because the `kar` will always set `vcpu_maxcpus = (host_cpus)/2` and I will met some error when one host have less cpus.

Reply:
Yeap, low chance to meet the value change problem, just logically speaking ~ it is up to you !